src/main/java/com/epam/esm/controller/UsersController.java

```diff
@@ -0,0 +1,91 @@
+public class UsersController {
+
+    private static final String JSON = "application/json";
+    private static final String ID = "id";
+    private static final String ALL_USERS = "users";
+    private static final String ALL_ = "users";
+
+    private final UserService service;
+
+    @Autowired
+    public UsersController(UserService service) {
+        this.service = service;
+    }
+
+    /**
+     * Returns an {@link ResponseEntity} object contained {@link HttpStatus} status and a {@link List} list of {@link User}
+     * users retrieved from database.
+     *
+     * @param page     specifies page of the result list.
+     * @param pageSize specifies number of tags to be displayed in a single page. In case page were passed without page size
+     *                 default page size applies.
+     * @return {@link ResponseEntity} contained both {@link HttpStatus} status and {@link List} of {@link User} users.
+     */
+    @GetMapping(produces = JSON)
+    public ResponseEntity<CollectionModel<UserDto>> getUsers(@RequestParam(required = false) Integer page,
+                                                             @RequestParam(required = false) Integer pageSize) {
+        List<UserDto> users = service.getUsers(page, pageSize);
+        users.forEach(user -> user.add(linkTo(methodOn(UsersController.class).getUser(user.getId())).withSelfRel()));
+        return ResponseEntity.ok(CollectionModel.of(users));
+    }
+
+    /**
+     * Returns an {@link ResponseEntity} object contained {@link HttpStatus} status and a {@link User} object from database.
+     *
+     * @param id id of {@link User} that has to be retrieved from database.
+     * @return {@link ResponseEntity} contained both {@link HttpStatus} status and an {@link User} object.
+     */
+    @GetMapping(value = "/{id}", produces = JSON)
+    public ResponseEntity<UserDto> getUser(@PathVariable(ID) long id) {
+        UserDto user = service.getUser(id);
+        user.add(linkTo(methodOn(UsersController.class).getUser(id)).withSelfRel());
+        user.add(linkTo(methodOn(UsersController.class).getUsers(4, 4)).withRel(ALL_USERS));
+        return new ResponseEntity<>(user, HttpStatus.OK);
+    }
+
+    /**
+     * Returns an {@link ResponseEntity} object contained {@link HttpStatus} status and a list of {@link Order} orders of specified user.
+     *
+     * @param id       id of {@link User} which orders have to be retrieved from database.
+     * @param page     specifies page of the result list.
+     * @param pageSize specifies number of tags to be displayed in a single page. In case page were passed without page size
+     *                 default page size applies.
+     * @return {@link ResponseEntity} contained both {@link HttpStatus} status and a list of {@link Order} orders of specified user
+     */
+    @GetMapping(value = "/{id}/orders", produces = JSON)
+    public ResponseEntity<CollectionModel<OrderDto>> getUsersOrders(@PathVariable(ID) long id,
```

Comment:
Why is this method located here? We're working with order resource here

Reply:
relocated them. it's now in the order's controller